client/components/data/query-domain-owner-username/index.ts

```diff
@@ -0,0 +1,41 @@
+import { useSelector } from 'react-redux';
+import { useQuerySitePurchases } from 'calypso/components/data/query-site-purchases';
+import useUsersQuery from 'calypso/data/users/use-users-query';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+import type { InfiniteData } from 'react-query';
+
+type User = {
+	login: string;
+};
+
+type UsersData = {
+	users: User[];
+};
+
+export function useDomainOwnerUserName(
+	selectedSite: SiteDetails | null | undefined,
+	domain: ResponseDomain | null | undefined
+) {
+	useQuerySitePurchases( selectedSite?.ID ?? -1 );
+
+	const purchases = useSelector( ( state ) => getSitePurchases( state, selectedSite?.ID ) );
+
+	const domainSubscription = purchases.filter(
+		( purchase ) => purchase.id === parseInt( domain?.subscriptionId ?? '0' )
+	)[ 0 ];
+
+	const fetchOptions = {
+		search: domainSubscription?.userId,
+		search_columns: [ 'ID' ],
+	};
```

Comment:
Shouldn't we return early whenever some mandatory data is not available?

```suggestion
	if ( ! domainSubscription ||| ! selectedSite ) {
		return '';
	}
	
	const fetchOptions = {
		search: domainSubscription?.userId,
		search_columns: [ 'ID' ],
	};
```

Reply:
I did it that way as we shouldn't make conditional "useEffects". It ends with complaining, but I will check again.

Reply:
That's it:

![image](https://user-images.githubusercontent.com/5689927/192333761-f643adbc-523f-4ddc-adde-712addf6f208.png)

Reply:
Oh, that's right.